system/jlib/jutil.cpp

```diff
@@ -2845,6 +2846,21 @@ int getEnum(const char *v, const EnumMapping *map)
     return 0;
 }
 
+int getEnum(const char *v, const EnumMapping *map, const char * type)
+{
+    if (v && *v)
+    {
+        while (map->str)
+        {
+            if (stricmp(v, map->str)==0)
+                return map->val;
+            map++;
+        }
+        throw makeStringExceptionV(JLIBERR_UnexpectedValue, "Unexpected %s '%s'", type, v);
+    }
+    return 0;
```

Comment:
@ghalliday This getNum() returns 0 if the search value is nullptr or empty string and that value may be confused with FOscope - FOscope has a enumeration value of 0.  May be return FOunknown if search value is nullptr or unknown?  Also, the function throws an exception of the value is not found.  This is inconsistant with the behaviour of getEnumText() and getEnum() - I feel the behaviour should be similar as the functions are in the same 'group' of functions.  I'd suggest moving throwing exception to the calling function.